fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -808,10 +808,11 @@ ACTOR Future<Void> releaseResolvingAfter(ProxyCommitData* self, Future<Void> rel
 // Commit one batch of transactions trs
 ACTOR Future<Void> commitBatch(
 	ProxyCommitData* self,
-	vector<CommitTransactionRequest> trs,
+	vector<CommitTransactionRequest>* pTrs,
 	int currentBatchMemBytesCount)
 {
 	//WARNING: this code is run at a high priority (until the first delay(0)), so it needs to do as little work as possible
+	state std::vector<CommitTransactionRequest> trs(std::move(*(const_cast<std::vector<CommitTransactionRequest>*>(pTrs))));
```

Comment:
I just saw this but why making a copy here? Cannot it simply use the pointer/reference passed in?

Reply:
It's not making a copy, it's transferring ownership via std::move. If you don't transfer the ownership then the lifetime of `*pTrs` ends too early.

Reply:
Ah you are right. Totally missed the fact that commitBatch is an actor.

Reply:
The original `trs`, which is a `const std::vector`, is *copied* to a local `trs` and we operate on the local one in `commitBatch`. The original `trs` is then never used, so a `move` semantic seems more reasonable in this case.